src/servers/Server.cc

```diff
@@ -206,10 +203,7 @@ Server::clientWriteDone(const CommIoCbParams &io)
 
     if (io.flag) {
         debugs(33, 2, "bailing after a write failure: " << xstrerr(io.xerrno));
-        LogTagsErrors lte;
-        lte.timedout = io.xerrno == ETIMEDOUT;
-        lte.aborted = !lte.timedout; // intentionally true for zero io.xerrno
-        terminateAll(Error(ERR_WRITE_ERROR, SysErrorDetail::NewIfAny(io.xerrno)), lte);
+        terminateAll(Error(ERR_WRITE_ERROR, SysErrorDetail::NewIfAny(io.xerrno)), LogTagsErrors::FromErrno(io.xerrno));
         return;
```

Comment:
Is it possible to do something about ERR_READ_TIMEOUT case (e.g., make sure it always has ETIMEDOUT) so that we can refactor FwdState::updateAleWithFinalError() to use the new method as well?

Reply:
I do not disagree, but LogTagsErrors do need to signal a TIMEOUT suffix for _both_ kinds of timeouts. Should FwdState::updateAleWithFinalError() do something like the following to simplify that code logic and reduce its duplication?

```C++
const auto lte = LogTagsErrors::FromErrno(err->type == ERR_READ_TIMEOUT ? ETIMEOUT : err->xerrno);
```

The above sketch effectively removes the ERR_NONE check in the existing code. I could not quickly figure out why we added that part of the check in recent commit 21f9015. Can you?

Reply:
> why we added that part of the check in recent commit https://github.com/measurement-factory/squid/commit/21f90159029aeb05fca55a6f48f120afb5f2b705

No, I could not find an explanation. It was added in the initial PR186 commit. Maybe some of ErrorState() creators called constructed it with ERR_NONE then? I looked through these callers now and could not find any. Anyway, I think it is better to reduce the code duplication as you suggested.